x-pack/plugins/enterprise_search/public/applications/app_search/components/search_ui/components/search_ui_form.tsx

```diff
@@ -0,0 +1,205 @@
+  { defaultMessage: 'Click to select' }
+);
+const SORT_FIELD_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.sortFieldLabel',
+  { defaultMessage: 'Sort fields (Optional)' }
+);
+const SORT_FIELD_HELP_TEXT = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.sortHelpText',
+  { defaultMessage: 'Used to display result sorting options, ascending and descending' }
+);
+const SORT_FIELD_PLACEHOLDER = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.sortFieldPlaceholder',
+  { defaultMessage: 'Click to select' }
+);
+const URL_FIELD_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.urlFieldLabel',
+  { defaultMessage: 'URL field (Optional)' }
+);
+const URL_FIELD_HELP_TEXT = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.urlFieldHelpText',
+  { defaultMessage: "Used as a result's link target, if applicable" }
+);
+const URL_FIELD_PLACEHOLDER = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.urlFieldPlaceholder',
+  { defaultMessage: 'Select a URL field' }
+);
+const GENERATE_PREVIEW_BUTTON_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.generatePreviewButtonLabel',
+  { defaultMessage: 'Generate a Preview' }
+);
+
+interface Option<T> extends Omit<EuiComboBoxOptionOption<T>, 'value'> {
+  value: string;
+}
+
+export const SearchUIForm: React.FC = () => {
+  const {
+    validFields,
+    validSortFields,
+    validFacetFields,
+    titleField,
+    urlField,
+    facetFields,
+    sortFields,
+  } = useValues(SearchUILogic);
+  const { engineName } = useValues(EngineLogic);
+  const {
+    onActiveFieldChange,
+    onFacetFieldsChange,
+    onSortFieldsChange,
+    onTitleFieldChange,
+    onUrlFieldChange,
+  } = useActions(SearchUILogic);
+
+  const previewHref = generatePreviwUrl(
+    {
+      fromKibana: 'true',
+      titleField,
+      urlField,
+      facets: facetFields,
+      sortFields,
+    },
+    engineName
+  );
+
+  const formatSelectOption = (fieldName: string) => {
+    return { text: fieldName, value: fieldName };
+  };
+  const formatMultiOptions = (fieldNames: string[]) =>
+    fieldNames.map((fieldName) => ({ label: fieldName, text: fieldName, value: fieldName }));
+  const formatMultiOptionsWithEmptyOption = (fieldNames: string[]) => [
+    { label: '', text: '', value: '' },
+    ...formatMultiOptions(fieldNames),
+  ];
+
+  const optionFields = formatMultiOptionsWithEmptyOption(validFields);
+  const sortOptionFields = formatMultiOptions(validSortFields);
+  const facetOptionFields = formatMultiOptions(validFacetFields);
+  const selectedTitleOption = formatSelectOption(titleField);
+  const selectedURLOption = formatSelectOption(urlField);
+  const selectedSortOptions = formatMultiOptions(sortFields);
+  const selectedFacetOptions = formatMultiOptions(facetFields);
+
+  return (
+    <div>
```

Comment:
I don't think we need this extra `div` wrapper

```suggestion
```

Reply:
c8f74b635c3b620d273c5ec2ed46273109f18db2